EMULATOR/LIBGPU.C

```diff
@@ -600,20 +717,21 @@ void ParseLinkedPrimitiveList(unsigned int packetStart, unsigned int packetEnd)/
 
 			Emulator_GenerateVertexArrayQuad(&g_vertexBuffer[g_vertexIndex], &poly->x0, &poly->x1, &poly->x3, &poly->x2, -1, -1);
 			Emulator_GenerateTexcoordArrayZero(&g_vertexBuffer[g_vertexIndex], 0);
-			Emulator_GenerateColourArrayQuad(&g_vertexBuffer[g_vertexIndex], &poly->r0,	NULL, NULL, NULL);
+			Emulator_GenerateColourArrayQuad(&g_vertexBuffer[g_vertexIndex], &poly->r0, NULL, NULL, NULL);
 
 			MakeTriangle();
 
 			g_vertexIndex += 6;
-			currentAddress += sizeof(POLY_F4);
-#if defined(DEBUG_POLY_COUNT)
+			primitive_size = sizeof(POLY_F4);
+	#if defined(DEBUG_POLY_COUNT)
 			polygon_count++;
-#endif
+	#endif
 			break;
 		}
 		case 0x2C:
 		{
 			POLY_FT4* poly = (POLY_FT4*)pTag;
+			activeDrawEnv.tpage = poly->tpage;
```

Comment:
Any reason behind this change? maybe it's a hack im unsure?

Reply:
I made this changes because Driver 2 uses *null* POLY_FT4 + SPRT to display textured polygon. It's weird but it works. It's better to use DR_TPAGE for that, but Reflections made it just like that